distmv/normal.go

```diff
@@ -75,9 +75,20 @@ func NewNormal(mu []float64, sigma mat64.Symmetric, src *rand.Rand) (*Normal, bo
 // ConditionNormal returns {nil, false} if there is a failure during the update.
 // Mathematically this is impossible, but can occur with finite precision arithmetic.
 func (n *Normal) ConditionNormal(observed []int, values []float64, src *rand.Rand) (*Normal, bool) {
+	if len(observed) == 0 {
+		panic("normal: no observed values")
+	}
 	if len(observed) != len(values) {
 		panic("normal: input slice length mismatch")
 	}
+	if len(observed) == n.Dim() {
+		panic("normal: all dimensions observed")
```

Comment:
This is not necessarily true. If I have n{0} then I have not observed all dimensions, but this will tell me I have. You can invoke the pigeon hole principle and say that all have been observed or at least one has been observed more than once. If the check is done below the de-duplication below you can use this message. Probably just weaken the text.